happytransformer/gen/trainer.py

```diff
@@ -0,0 +1,79 @@
+"""
+
+Fine-tuning for text generation odels.
+
+Based on the tutorial found here:
+https://github.com/huggingface/transformers/blob/master/examples/pytorch/language-modeling/run_clm.py
+"""
+import json
+from transformers import default_data_collator
+from happytransformer.happy_trainer import HappyTrainer, EvalResult
+from happytransformer.fine_tuning_util import preprocess_concatenate
+from datasets import load_dataset
+
+class GENTrainer(HappyTrainer):
+    """
+    Trainer class for HappyWordPrediction
+    """
+
+    def train(self, input_filepath, args):
+
+        if not args["load_data"]:
+            self.logger.info("Preprocessing dataset...")
+            datasets = load_dataset("text", data_files={"train": input_filepath})
+            tokenized_datasets = preprocess_concatenate(self.tokenizer, datasets, args, False)
+
+        else:
+            self.logger.info("Loading dataset from %s...", args["load_data_path"])
+            tokenized_datasets = load_dataset("json", data_files={"train": args["load_data_path"]}, field='train')
+
+        if args['save_data']:
+            if args['load_data']:
+                self.logger.warning("Both save_data and load_data are enabled,")
+
+            self.logger.info("Saving training dataset to %s...", args["save_data_path"])
+
+            self._generate_json(args['save_data_path'], tokenized_datasets["train"], "train")
+
+        self.logger.info("Training...")
+
+        self._run_train(tokenized_datasets['train'], args, default_data_collator)
+
+    def eval(self, input_filepath, args):
+
+        if not args["load_data"]:
```

Comment:
Surely this flag should be called `bypass_preprocessing`

Reply:
hhhmmm. 

I think "bypass_preprocessing" might sound too confusing/technical and doesn't flow well with "save_data." 

Whatever we use for the flag should match the path variable. "bypass_preprocessing_flag" may be too long. 


If we were to use "bypass_preprocessing" then the "save" counterpart would be something like  "save_preprocessing_data." 

I'm fully open to suggestions.

Reply:
What kind of data do `load_data` and `save_data` control exactly?

Reply:
The inputted eval data, which is a text file, gets preprocessed by going through concatenation and tokenization.  "save_data" saves the final result of this preprocessing process to the path given by "save_data_path". Then, by setting  "load_data"  to True and providing a path for "load_data_path," the output from using save_data can be used to bypass preprocessing -- thus reducing compute.

Reply:
I agree with the sentiment but the naming should make the fact that these flags affect preprocessing only should be explicit. We don't want users to think that these flags affect saving of predictions

Reply:
> I agree with the sentiment but the naming should make the fact that these flags affect preprocessing only should be explicit. We don't want users to think that these flags affect saving of predictions

very good suggestion

Reply:
0dd2078

Reply:
I used "save_preprocessed_data" and "load_preprocessed_data"